File: src/model/statistics.py
class DailyFreq:
    def __init__(
            self,
            date=0,
            freq=0) -> None:
        self.date = date
        self.freq = freq

    def to_json(self):
        return {
            "date": self.date,
            "freq": self.freq
        }

class Statistics:
    def __init__(
            self,
            uid=0,
            overheat_num=0,
            normal_num=0,
            underheat_num=0,
            processing_num=0,
            daily_freqs:list[DailyFreq]=[]) -> None:
        self.uid = uid
        
        self.overheat_num = overheat_num
        self.normal_num = normal_num
        self.underheat_num = underheat_num
        self.processed_num = overheat_num + normal_num + underheat_num
        self.processing_num = processing_num
        self.total_num = self.processed_num + self.processing_num
        self.abnormal_num = overheat_num + underheat_num
        
        self.overheat_rate : float
        self.normal_rate : float
        self.underheat_rate : float
        self.abnormal_rate : float
        if self.processed_num > 0:
            self.overheat_rate = float(overheat_num) / float(self.processed_num)
            self.normal_rate = float(normal_num) / float(self.processed_num)
            self.underheat_rate = float(underheat_num) / float(self.processed_num)
            self.abnormal_rate = float(self.abnormal_num) / float(self.processed_num)

        self.daily_freqs : list[DailyFreq] = daily_freqs
        
    def to_json(self):
        return {
            "uid": self.uid,
            "overheat_num": self.overheat_num,
            "normal_num": self.normal_num,
            "underheat_num": self.underheat_num,
            "processing_num": self.processing_num,
            "abnormal_num": self.abnormal_num,
            "overheat_rate": self.overheat_rate,
            "normal_rate": self.normal_rate,
            "underheat_rate": self.underheat_rate,
            "abnormal_rate": self.abnormal_rate,
            "daily_freqs:": [daily_freq.to_json() for daily_freq in self.daily_freqs]
        }

File: ui/app/src/main/java/com/luozi/fireeyewatcher/model/Statistics.java
package com.luozi.fireeyewatcher.model;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.List;

public class Statistics {

    public static class DailyFreq {
        public int date;
        public int freq;

        public DailyFreq(int date, int freq) {
            this.date = date;
            this.freq = freq;
        }

        public static DailyFreq parseFromJson(JSONObject json) throws JSONException {
            return new DailyFreq(
                    json.getInt("date"),
                    json.getInt("freq")
            );
        }
    }

    public int userId;
    public int overheatNum;
    public int normalNum;
    public int underheatNum;
    public int processingNum;
    public int abnormalNum;
    public double overheatRate;
    public double normalRate;
    public double underheatRate;
    public double abnormalRate;
    public List<DailyFreq> dailyFreqs;

    public Statistics(
            int userId, int overheatNum, int normalNum,
            int underheatNum, int processingNum, int abnormalNum,
            double overheatRate, double normalRate, double underheatRate,
            double abnormalRate, List<DailyFreq> dailyFreqs
    ) {
        this.userId = userId;
        this.overheatNum = overheatNum;
        this.normalNum = normalNum;
        this.underheatNum = underheatNum;
        this.processingNum = processingNum;
        this.abnormalNum = abnormalNum;
        this.overheatRate = overheatRate;
        this.normalRate = normalRate;
        this.underheatRate = underheatRate;
        this.abnormalRate = abnormalRate;
        this.dailyFreqs = dailyFreqs;
    }

    public static Statistics parseFromJson(JSONObject json) throws JSONException {
        int userId = json.getInt("uid");
        int overheatNum = json.getInt("overheat_num");
        int normalNum = json.getInt("normal_num");
        int underheatNum = json.getInt("underheat_num");
        int processingNum = json.getInt("processing_num");
        int abnormalNum = json.getInt("abnormal_num");
        double overheatRate = json.getDouble("overheat_rate");
        double normalRate = json.getDouble("normal_rate");
        double underheatRate = json.getDouble("underheat_rate");
        double abnormalRate = json.getDouble("abnormal_rate");

        JSONArray freqs = json.getJSONArray("daily_freqs");
        List<DailyFreq> dailyFreqs = new ArrayList<>();
        for (int i = 0; i < freqs.length(); i++) {
            DailyFreq dailyFreq = DailyFreq.parseFromJson(freqs.getJSONObject(i));
            dailyFreqs.add(dailyFreq);
        }

        return new Statistics(
                userId, overheatNum, normalNum, underheatNum,
                processingNum, abnormalNum, overheatRate,
                normalRate, underheatRate, abnormalRate,
                dailyFreqs
        );
    }
}
